src/Microsoft.AspNet.Hosting/WebHostBuilder.cs

```diff
@@ -163,6 +166,17 @@ public WebHostBuilder UseEnvironment(string environment)
             return this;
         }
 
+        public WebHostBuilder UseApplication(IHttpApplication application)
```

Comment:
Let's not go there yet. If we were going to do this you'd want to bypass running Startup completely.